src/main/kotlin/slack/cli/sarif/SarifUtil.kt

```diff
@@ -122,7 +122,16 @@ internal fun List<SarifSchema210>.merge(
     flatMap { it.runs.single().results.orEmpty() }
       // Some projects produce multiple reports for different variants, so we need to
       // de-dupe.
-      .distinct()
+      // Using the default distinct() function leaves duplicates, so using a custom selector
+      .distinctBy {
+        it.ruleID +
+          it.message +
+          it.locations?.first()?.physicalLocation?.artifactLocation?.uri +
+          it.locations?.first()?.physicalLocation?.region?.startLine +
+          it.locations?.first()?.physicalLocation?.region?.startColumn +
+          it.locations?.first()?.physicalLocation?.region?.endColumn +
+          it.locations?.first()?.physicalLocation?.region?.endLine
+      }
```

Comment:
Could you use `it.shallowHash`? It uses the same heuristics. It's currently missing `message`, but that's a bug and you could add it here.